### ISSUE TITLE
Determine how to report a development version of ClassicPress

### ISSUE THREAD

**Issue by nylen** [ClassicPress/ClassicPress#58]
The following code is a bit broken at the moment:

https://github.com/ClassicPress/ClassicPress/blob/6a6ce589f760851f10654349b9a181091f495642/src/wp-includes/default-constants.php#L84-L94

`$wp_version` is used in this line:

```
$develop_src = false !== strpos( $GLOBALS['wp_version'], '-src' );
```

We should come up with a better (ideally totally automatic) way of determining whether ClassicPress is being run from a development install, and reporting this information and setting the `SCRIPT_DEBUG` flag.

**Comment by stefanos82** [comment]
@nylen Wouldn't this cause problems, if for example I fetch the latest repo updates?

How do I know which dev version am currently using?

Wouldn't make much more sense to have something like, for `git rev-parse --short=7 HEAD` and get a 7-character hash appended to `$GLOBALS['wp_version']` instead?

```php
$repo_hashval = exec('git rev-parse --short=7 HEAD'); // assigns something like '97ddf8c'
$develop_src = false !== strpos( $GLOBALS['wp_version'], "-$repo_hashval" );
```

**Comment by nylen** [comment]
Yes, I'm thinking of doing something like this. We should probably do it by reading files only, rather than `exec`, since this is disabled in some environments.

The dev version information should also go in `$cp_version` instead.

**Comment by Mte90** [comment]
Maybe we can do more easily do a `.version` file as example that is available only on git and when we do the build package that file doesn't exist. 
So checking if the file exist will say to classicpress that is the dev version.

**Comment by stefanos82** [comment]
Yep that would be really cool and easy to implement.

@nylen what confuses me with your "or `git` may not be available" statement is, how is ClassicPress suppose to distinguish a stable `.zip` file from a development version?

Are we supposed to generate a dev edition on the fly, much like how GitHub is letting us do so?

**Comment by nylen** [comment]
I mean that we should read the current commit from the `.git` directory using filesystem operations.

We can apply whatever transformation we want to the build version, that is easy. Maintaining a `.version` file for development installs is not so easy: how do we make sure that it gets updated with each commit?

**Comment by ginsterbusch** [comment]
I'd just use release tags for regular versions, and the commit hash for development versions.
But the actual update check is powered a) by the wp.org API and b) the $db_version. If the $db_version is different - update commences. Et voila! All done.

cu, w0lf.

**Comment by ginsterbusch** [comment]
Addendum: WP lets you decide whether you're using a stable or a development version. So the switch happens first via wp.org API.

cu, w0lf.

**Comment by Mte90** [comment]
I think that wordpress use something svn side that on push update the version, maybe we can do something like that?
Or to be more easy we can show the date of the last update of the build instead of the commit so this can be more easy to change.

**Comment by stefanos82** [comment]
Can't we do this via `Gruntfile.js`? It seems it does so in `version`.

**Comment by ginsterbusch** [comment]
Like I said before: Just use anything straight from the `src` directory for an existing install, and for a new one, copy the `wp-config-sample.php` from the repository root directory into your install directory. Done :)

**Comment by nylen** [comment]
Aiming to get this done before our first beta release.

**Comment by nylen** [comment]
I've got a PR out to fix the original issue I reported here (#153).  We need to go ahead and fix it, because I accidentally made it worse in #147.

There, we changed the version string marker in the source repository from `-src` to `+dev`.  The `+` is semver-compliant and generally makes more sense as part of an emerging plan for versioning and telling the difference between builds and source installs.

> How do I know which dev version am currently using?

You do `git status`.  There should never be a situation where you're running a version with `+dev` and you don't have `git status` available - this setup is only for development against ClassicPress itself.

> how is ClassicPress suppose to distinguish a stable .zip file from a development version?

A stable build will have a version number like `1.0.0` or `1.0.0-beta1`.  A development version will always have the `+dev` suffix.  The full version number in the source tree will change every time we do a release, and for now, the progression is documented [in the code](https://github.com/ClassicPress/ClassicPress/blob/b325a7c5a41585d4df8f1e810921fd1557540646/src/wp-includes/version.php#L2-L30).

There are also nightly builds, which have a version number like `1.0.0-alpha0+nightly.20181020`.  They are now available [here](https://github.com/ClassyBot/ClassicPress-nightly/releases).
